### PR TITLE
[#47] Add automated tests to CI pipeline

### LINKED ISSUE #47: Automated testing in the pipeline
## Description
We need automated testing running in the pipeline

### Acceptance Criteria
- Unit tests run in the pipeline
- Integration tests run in the pipeline
- Any failures block merge

## Definition of Done
[Definition of Done](https://github.com/coforma/swift-tech-challenge/blob/main/documentation/product/definition_of_done.md)

### PR DESCRIPTION
**Pull request by braxex**
## Summary

We need to get our automated tests running in the CI pipeline. Any failures or regressions should block merges.

Fixes #47

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How To Test
Upon merge, pipeline should run automated tests, and any failures should block merge.

## Checklist:
- [x] My code follows the style guidelines of this project and has no linting errors
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, including any applicable ADRs
- [x] My changes generate no new warnings
- [x] I have added tests that fail without these changes
- [x] New and existing tests (unit, integration, accessibility) pass locally 
- [x] Documentation updated
- ~If there are security concerns they are addressed or ticketed after being discussed~

### CODE REVIEW

**Comment by braxex** [comment]
@cuevas1876 Wanted to chat with you about our Terraform deploys and see how I can effectively hook into those for our automated unit and integration tests. Blocked on this until we can figure that out. I'll ping you in the am. 👍

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      - name: 'Setup jq'
+        uses: dcarbone/install-jq-action@v2.1.0
+        with:
+          version: '${{ inputs.version }}'
+          force: '${{ inputs.force }}'
+      - name: 'Check jq'
+        # language=sh
+        run: |
+          which jq
+          jq --version
```
Comment (gmrabian) [inline .github/workflows/testing.yml:32]: where are we using jq?

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      - uses: actions/setup-node@v3
```
Comment (gmrabian) [inline .github/workflows/testing.yml:33]: nit: latest setup-node is v4

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        with:
+          node-version-file: ".nvmrc"
+      - uses: actions/cache@v3
```
Comment (gmrabian) [inline .github/workflows/testing.yml:36]: nit: could upgrade to cache v4 release

File: .github/workflows/testing.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            **/node_modules
+            ~/.cache/Cypress
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: set path
+        run: |
+          echo "PATH=$(pwd)/node_modules/.bin/:$PATH" >> $GITHUB_ENV
+      - name: run unit tests (frontend)
+        run: ./scripts/test-unit.sh src/app/testing/jest/frontend
+      - name: run unit tests (backend) src/app/testing/jest/frontend
+        run: ./scripts/test-unit.sh
```
Comment (gmrabian) [inline .github/workflows/testing.yml:48]: I think the script is overkill when we could just have a step for `yarn install` (and cache it) and a step for `yarn dev:cov`

File: utils/run-unit-tests.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (gmrabian) [inline utils/run-unit-tests.sh:1]: do we really need this script?

**Review by gmrabian — COMMENTED** [review]
nits and ideas for simplification

File: .github/workflows/testing.yml (hunk 4 above)
Comment (braxex) [reply .github/workflows/testing.yml]: Ooh very good point. Switching.

File: utils/run-unit-tests.sh (hunk 5 above)
Comment (braxex) [reply utils/run-unit-tests.sh]: Nope!

**Review by gmrabian — DISMISSED** [review]
super clean! love it

**Review by gmrabian — APPROVED** [review]

**Pull request merged**